.github/workflows/build.yml

```diff
@@ -0,0 +1,50 @@
+name: Lint, Build, E2E Test
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    name: Linter
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.32
+  test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.14.4'
+      - name: Run unit tests
+        run: make test
+  e2e-kubernetes:
+    name: Kuebrnetes E2E tests
+    needs: [ test ]
+    runs-on: ubuntu-18.04 # required by manusa/actions-setup-minikube@v2.0.1
+    env:
+      E2E_MINIKUBE_DRIVER: docker
+    strategy:
+      matrix:
+        k8s_version:
+          - v1.16.15
+          - v1.17.14
+          - v1.18.12
```

Comment:
Can we test on a similar matrix of k8s e2e tests?

Reply:
Agreed, it makes sense to use the same matrix.